src/main/kotlin/cacophony/semantic/syntaxtree/ASTGeneration.kt

```diff
@@ -142,6 +142,32 @@ private fun operatorRegexToAST(children: List<ParseTree<CacophonyGrammarSymbol>>
     }
 }
 
+private fun callAndFieldRegexToAst(children: List<ParseTree<CacophonyGrammarSymbol>>, diagnostics: Diagnostics): Expression {
+    val childNum = children.size
+    if (childNum == 1) {
+        return generateASTInternal(children[0], diagnostics)
+    } else {
+        val newChildren = children.subList(0, childNum - 1)
+        val range = Pair(first = children[0].range.first, second = children[childNum - 1].range.second)
+        val lhs = callAndFieldRegexToAst(newChildren, diagnostics)
+        val lastChild = children[childNum - 1]
+        return if (lastChild is ParseTree.Leaf && lastChild.token.category == VARIABLE_IDENTIFIER) { // field reference
+            if (lhs is Assignable) {
+                FieldRef.LValue(range, lhs, lastChild.token.context)
+            } else {
+                FieldRef.RValue(range, lhs, lastChild.token.context)
+            }
+        } else if (lastChild is ParseTree.Branch && lastChild.production.lhs == FUNCTION_CALL) {
+            val arguments = mutableListOf<Expression>()
+            assert(lastChild.production.lhs == FUNCTION_CALL)
+            for (child in lastChild.children) {
+                arguments.add(generateASTInternal(child, diagnostics))
+            }
```

Comment:
**suggestion:** this can be simplified
```suggestion
            assert(lastChild.production.lhs == FUNCTION_CALL)
            val arguments = lastChild.children.map { generateASTInternal(it, diagnostics) }
```